### DISCUSSION TITLE
V167 OBD2

### DISCUSSION THREAD

**Discussion by mschrader15** [mercedes-benz/foss#24] [Q&A] [answered]
Hey Y'all,

**Let me preface by saying this probably isn't the correct location or forum to ask this question in, but I found Daimler to be an active contributor to [cantools](https://github.com/eerimoq/cantools), and I am stumped**

I am a graduate student at the University of Alabama, where we intend to use a V167 (GLE350) to test CV2X and traffic light optimization. We want to log simple vehicle information via the OBD2 CAN bus, and so planned on using a [CANable](https://canable.io/) and open source python libraries ([cantools](https://github.com/eerimoq/cantools) & [python-can](https://github.com/hardbyte/python-can)) to write a simple logging script. This script worked great on a 2015 VW Jetta, but moving to the V167, we have had no luck. It simply doesn't respond to the **0x7DF** OBD2 message. We see lots of bus traffic, but none that correspond to the OBD2 standard message IDs.

Can you offer any guidance? 

Best,
Max

**Comment by mschrader15** [comment, marked as answer]
I answered my own question. The V167 uses extended CAN, so the OBD2 request message id is `0x18DB33F1`

**Comment by andlaus** [comment]
great that you like cantools, and we are very happy that you found the solution for the problem!
